apiserver/facades/client/highavailability/highavailability.go

```diff
@@ -451,3 +458,42 @@ func (api *HighAvailabilityAPI) ControllerDetails(
 	}
 	return results, nil
 }
+
+// NewHighAvailabilityAPI returns a new HighAvailabilityAPI instance.
+func NewHighAvailabilityAPI(
+	ctx context.Context,
+	st *state.State,
+	auth facade.Authorizer,
+	nodeService NodeService,
+	machineService MachineService,
+	applicationService ApplicationService,
+	modelInfoService ModelInfoService,
+	controllerConfigService ControllerConfigService,
+	networkService NetworkService,
+	blockCommandService BlockCommandService,
+	logger corelogger.Logger,
+) (*HighAvailabilityAPI, error) {
+	if !auth.AuthClient() {
+		return nil, apiservererrors.ErrPerm
+	}
+
+	modelInfo, err := modelInfoService.GetModelInfo(ctx)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if modelInfo.Type == model.CAAS {
+		return nil, errors.NotSupportedf("high availability on kubernetes controllers")
+	}
```

Comment:
This should still be done inside the facade registration code. The existing `newHighAvailabilityAPI` code can be tweaked to use the domain service to get the model type etc. The `AuthClient` check can remain in the register code. We then don't need this `NewHighAvailabilityAPI` func - we can stick with creating the struct directly in the register code. This would also avoid the unnecessary change to `schema.json`.

Reply:
fixed.